### DISCUSSION TITLE
Apex Syntax Highlighting Extension

### DISCUSSION THREAD

**Discussion by icarvagu** [zed-industries/zed#14785] [Feature Requests]
Hi,

I'm working on creating an extension to add syntax highlighting for Apex, but I'm encountering some issues. Since Apex is quite similar to Java with only minor differences, I based my approach on the Java extension.

I followed the Zed documentation and ran tree-sitter build --wasm to generate the WASM file, but I'm still having trouble with the installation process. The extension doesn't seem to install correctly but it clones and create the grammar paste.

Could anyone offer some guidance or troubleshooting tips?

Here is the repo with what i did: https://github.com/icarvagu/zed-apex (i did in two diferent ways, two branches)

Thanks!

**Comment by icarvagu** [comment]
I acctually figured out what is happenning:

Caused by:

`failed to compile apex parser with clang: clang: error: no such file or directory: '/Users/igorcguedes/Documents/Projeto/zed-apex/grammars/apex/src/parser.c'
`
How can I change the directory that TreeSitter looks in to find the parser? The default structure of the TreeSitter-Apex repository is slightly different from what I need. While I can manually adjust the structure, it gets reset when I clone the repository again. I'm considering cloning the repository and modifying the folder structure to make it work.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
